### DISCUSSION TITLE
Experience of Coordinators for ZHA and other systems.

### DISCUSSION THREAD

**Discussion by MattWestb** [zigpy/zigpy#576] [General]
I finding it interesting of other users experiences of coordinators in ZHA.
I have experience RspBee I but not in ZHA only in deCONZ. TI CC-2531 and IKEA Billy EZSP (One IKEA ICC-A-1 with Silabs EFR32 first gen).

I think for testing ZHA the CC-2531 is good for beginners that like testing Zigbee out but is very fast getting problem if adding mutch devices. TI have discontinuing all Zigbee products so no R23 update is possible in the future also for the newer chips and bug and security fixes is going to being one problem in near future. Its possible using it as one sniffer with sniffing firmware and also as router with router firmware (and some nice DYS firmwares with HA functions). Zigbee HA 1.2 is the most stable NCP firmware and Zigbee 3 is also supported but having problem with too less flash / ram / CPU for working well.
Community support for the firmware is nearly excellent but its very closed and is made thru patching then its not allowed releasing the source code.

The old TI chips is used in many little older devices that also having some strange problems like old OSRAM and newer "chinese Zigbee 3" devices and having bad radios / antennas and corrupting packages / missed ACKs.

RaspBee I Was the premium coordinator and the gen 2 should being better. The first gen of Rasp/CornBee have TouchLink implanted in the firmware and the second not that is one deal breaker for My. Both gens is sharing the same serial protocol and as long its not being broken boths is supported by ZHA. Router and sniffing firmware is released but is outdated and broken.
If I have understanding it right is it ideal for light systems then all grupes that is made by the NCP is automatically setted up for reporting the traffic. The module have normally very good ceramic antennas that is making the coordinator getting mutch devices directly connected and not using the other router in the mesh (good and bad).
The firmware is closed source but is aktive mentraned for the company.
The native applications i dot like comment here of religious reason!! 

EZSP: I dont knowing the performance of the EM3581 series but the first EFR32 have the same or better spec then Rasp/CornBee II both processor and flash / ram and equal radio / RF performance and the second gen EFR32 is much better. EZSP NCPs can acting as coordinator or router then being setted up and also as sniffer then the coordinator / router is offline so no need flashing separate sniffer firmware. The EZSP is aktive development and maintained from Silabs and its possible for the community making and distributing new firmware for free. The old EM family and first gen with lesser than 512K flash is discontinuing and don't getting Zigbee R23 support only bug and security fixes but is still on the edge version to day (12/2020). The chips is having one very good radio and have very good logick for receiving / transmitting and build in RF amplifier. The EZSP protocol loks being very low level but also very powerful that our devs have taking advantages of. I think one minus is that is only possible configuring 16 ZLL groups that the coordinator is listening to for getting the events in the system (correcting my if im wrong).
The most well known user of EZSP chips was IKEA that have using first gen EFR32 but all new  tuya devices is having first or second gens EFR32.

Do you have some more experience of thesis coordinators or NXP, Nordic and AMTEL based ones that can being used in ZHA then please share your experience her with our community.

**Comment by puddly** [comment]
> TI have discontinuing all Zigbee products so no R23 update is possible in the future also for the newer chips and bug and security fixes is going to being one problem in near future. 

Really? I was under the impression they're still actively supporting the SimpleLink SDK (CC2652R and the CC1352P).

**Reply by MattWestb** [reply]
I have not seeing the offishal acuntmens from TI but i have reading comments that is discontinuing on Z2M git. Perhaps its only the old CC-253X that is end of life ?
By the way its different of supporting one aktive or discontinued product or is still active development of the product. If active development is very likely getting some old bugs backported that is being found later then updating / implanting new functions in the code and very unlikely if its inactive / discontinued.
Like Silbs is discontinuing first gen EFR32 (with expectations) but still supporting it active until real end of life of it. So no new major things is coming for the discontinuing products but still for all active ones like Zigbee R23 for the Silabs NCPs.

**Reply by puddly** [reply]
You can see the currently supported releases here: https://www.ti.com/tool/Z-STACK

 - Z-Stack 3.0.2 is "active" but has not been updated by TI since October 2018. It runs on the CC253X chips.
 -  SIMPLELINK-CC13X2-26X2-SDK (Z-Stack 3.30+, 4.x) runs on the newer chips and was last updated two months ago..

**Reply by MattWestb** [reply]
So Z-Stack 3.30+, 4.x is active development and CC-253x is not but still active supported until end of life.

Is the SIMPLELINK getting Zigbee R23 support or staying with R22 and butfixing ?

**Reply by puddly** [reply]
I would imagine it would but I do not know for sure.

**Reply by MattWestb** [reply]
For my its no problems the TIs ZNCP is porfing to working with Zigbee HA 1.2 and newer and is enough for getting one system running and some is doing it more then well :-)))
And on the bug side you have seeing the EZSP one that is bombing IKEA devices and all stacks is having more or less of them (bugs).

**Reply by MattWestb** [reply]
Its looks like touch link is broken in the last releases of Z-Stack for the new gen NCPs.
The old ones looks working OK but TI have don some bad things in the SDK. Z2M is having problem with it but is waiting for TI to fixing the code.
No problem for the moment then ZHA dont have implantation of touch link (exception is HUE thief that using one old bellows but the python is outdated and not working to day).

**Reply by MattWestb** [reply]
@puddly I think you is having right 3.0.X is ended but apparently not the 3.X.0 = new TI hardware.
One interesting bug in the old CC-253X and CC2538: [CC2531 - ZB 3.0, panid/epanid issues](https://github.com/Koenkk/zigbee2mqtt/issues/4086#) I dont knowing if you have seen it being triggered in the TI NCPs but can being good to knowing it if its popping up.
If its showing up you can dumping the NCPs NVR and analysing it that Z2M is not doing (also they dont resetting the NCPs NVR if having problems they reflashing them) = dont have deep knowledge of the function of the device.

**Comment by pipiche38** [comment]
> Do you have some more experience of thesis coordinators or NXP, Nordic and AMTEL based ones that can being used in ZHA then please share your experience her with our community.


ZiGate is based on NXP!

**Reply by MattWestb** [reply]
Yes i knowing but i dont  have any experience of ZiGate / NXP coordinator ;-(
Feel more then free to writing of your experience of the coordinator in ZHA.
All experience is very welcome.

**Reply by MattWestb** [reply]
I finding it great that is possible reusing devices and reflashing them with one coordinator firmware and transforming it on one ZNCP like the ZiGate is /can doing !
I is doing the same with IKEA devices and flashing EZSP firmware and it working great :-)))
